test/external-modules/multi-project/src/javaRestTest/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsMultiProjectIT.java

```diff
@@ -0,0 +1,165 @@
+import org.elasticsearch.multiproject.MultiProjectRestTestCase;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.cluster.local.distribution.DistributionType;
+import org.elasticsearch.test.rest.ObjectPath;
+import org.junit.ClassRule;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+
+public class IndicesStatsMultiProjectIT extends MultiProjectRestTestCase {
+
+    private static final String PASSWORD = "hunter2";
+
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
+        .nodes(1)
+        .distribution(DistributionType.INTEG_TEST)
+        .module("test-multi-project")
+        .setting("test.multi_project.enabled", "true")
+        .setting("xpack.security.enabled", "true")
+        .user("admin", PASSWORD)
+        .build();
+
+    @Override
+    protected String getTestRestCluster() {
+        return cluster.getHttpAddresses();
+    }
+
+    @Override
+    protected Settings restClientSettings() {
+        final String token = basicAuthHeaderValue("admin", new SecureString(PASSWORD.toCharArray()));
+        return Settings.builder().put(ThreadContext.PREFIX + ".Authorization", token).build();
+    }
+
+    public void testIndicesStats() throws IOException {
+        // Create two projects. We will use the default project as the third project in this test.
+        createProject("project-1");
+        createProject("project-2");
+
+        // Create and write data into a number of indices.
+        // Some of the index names are used only in one project, some in two projects, some in all three.
+        int numDocs1Only = createPopulatedIndex("project-1", "my-index-project-1-only");
+        int numDocs2Only = createPopulatedIndex("project-2", "my-index-project-2-only");
+        int numDocsDefaultOnly = createPopulatedIndex("default", "my-index-default-project-only");
+        int numDocs1Of1And2 = createPopulatedIndex("project-1", "my-index-projects-1-and-2");
+        int numDocs2Of1And2 = createPopulatedIndex("project-2", "my-index-projects-1-and-2");
+        int numDocs2Of2AndDefault = createPopulatedIndex("project-2", "my-index-projects-2-and-default");
+        int numDocsDefaultOf2AndDefault = createPopulatedIndex("default", "my-index-projects-2-and-default");
+        int numDocs1All = createPopulatedIndex("project-1", "my-index-all-projects");
+        int numDocs2All = createPopulatedIndex("project-2", "my-index-all-projects");
+        int numDocsDefaultAll = createPopulatedIndex("default", "my-index-all-projects");
+
+        // Check indices stats for project 1.
+        Map<String, Object> statsForProject1 = getAsOrderedMapInProject("/_stats", "project-1");
+        assertThat(ObjectPath.evaluate(statsForProject1, "_all.total.docs.count"), equalTo(numDocs1Only + numDocs1Of1And2 + numDocs1All));
```

Comment:
Nit: I think you can make `getAsOrderedMapInProject` return an `ObjectPath` by doing
```java
private static ObjectPath getAsObjectPathInProject(String endpoint, String projectId) throws IOException {
        return ObjectPath.createFromResponse(client().performRequest(setRequestProjectId(new Request("GET", endpoint), projectId)));
    }
```
and then you can do
```suggestion
        ObjectPath statsForProject1 = getAsObjectPathInProject("/_stats", "project-1");
        assertThat(statsForProject1.evaluate("_all.total.docs.count"), equalTo(numDocs1Only + numDocs1Of1And2 + numDocs1All));
```
(and we even have `ObjectPath#evaluateMapKeys` which you can use below)
As I said, it's a nit pick, it's just a little bit more compact.